File: haskell/anagram/Anagram.hs
module Anagram(anagramsFor, isAnagram) where

import Data.Char(toLower)
import Data.List(sort)

anagramsFor :: String -> [String] -> [String]
anagramsFor word = filter (isAnagram word)

isAnagram :: String -> String -> Bool
isAnagram a b
    | lowerA == lowerB = False
    | otherwise = sort lowerA == sort lowerB
    where
        lowerA = map toLower a
        lowerB = map toLower b

File: python/anagram/anagram.py
def detect_anagrams(word, input_list):
    return [anagram for anagram in input_list if is_anagrams(word, anagram)]


def is_anagrams(A, B):
    lowerA = A.lower()
    lowerB = B.lower()
    if lowerA == lowerB:
        return False

    sortedA = sorted(lowerA)
    sortedB = sorted(lowerB)
    if sortedA == sortedB:
        return True
    return False
